### PR TITLE
Use old style variable annotation

### LINKED ISSUE #109: Manylinux build crashes due to python syntax error in wrap
Hi @varunagrawal and @ProfFan , I think some recent wrap changes broke the wheel-builder in `gtsam-manylinux`. Here's the error message:

```
Scanning dependencies of target pybind_wrap_gtsam
[  7%] Building C object gtsam/3rdparty/metis/libmetis/CMakeFiles/metis-gtsam.dir/__/GKlib/itemsets.c.o
[  7%] Generating gtsam.cpp
[  7%] Building C object gtsam/3rdparty/metis/libmetis/CMakeFiles/metis-gtsam.dir/__/GKlib/mcore.c.o
Traceback (most recent call last):
  File "/gtsam/wrap/scripts/pybind_wrap.py", line 13, in <module>
    import gtwrap.interface_parser as parser
  File "/gtsam/wrap/gtwrap/interface_parser/__init__.py", line 17, in <module>
    from .classes import *
  File "/gtsam/wrap/gtwrap/interface_parser/classes.py", line 18, in <module>
    from .function import ArgumentList, ReturnType
  File "/gtsam/wrap/gtwrap/interface_parser/function.py", line 58
    self.parent: Union[ArgumentList, None] = None
               ^
SyntaxError: invalid syntax
make[2]: *** [python/gtsam.cpp] Error 1
make[1]: *** [python/CMakeFiles/pybind_wrap_gtsam.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

Could yall take a quick look when you get a chance?

Link to run:
https://github.com/borglab/gtsam-manylinux-build/runs/2802606418?check_suite_focus=true

Link to all runs:
https://github.com/borglab/gtsam-manylinux-build/actions/runs/928431417

### PR DESCRIPTION
**Pull request by varunagrawal**
Fixes #109

### CODE REVIEW

File: gtwrap/interface_parser/function.py [hunk 1]
```diff
@@ -47,7 +47,7 @@ def __init__(self,
             self.ctype = ctype
         self.name = name
         self.default = default
-        self.parent: Union[ArgumentList, None] = None
+        self.parent = None  # type: Union[ArgumentList, None]
```
Comment (johnwlambert) [inline gtwrap/interface_parser/function.py:50]: Thanks Varun. A few questions:

1. We are removing the first type annotation. But there are 10 more below. So is it really a fix?
2. I can't for the life of me figure out where Python 2 is being used in the wheel builder. I assume any Python 3 can support type hints. I would prefer to just upgrade the python version. Python2 is dead to me.

Comment (varunagrawal) [reply gtwrap/interface_parser/function.py:50]: 1. Python 3.5 doesn't support variable annotations but it supports function annotations. More [here](https://www.python.org/dev/peps/pep-0526/).
2. It is being set in the `.github` directory. We don't support Python2 here so no need to worry about that. :slightly_smiling_face:

Comment (johnwlambert) [reply gtwrap/interface_parser/function.py:50]: Thanks Varun. Do we still need support for Python 3.5?

Comment (varunagrawal) [reply gtwrap/interface_parser/function.py:50]: We should until there is a strong reason not to? Some organizations may not have upgraded.

**Review by johnwlambert — APPROVED** [review]

**Pull request merged**